sys/src/9/ufs/ffs_vfsops.c

```diff
@@ -1629,7 +1632,8 @@ ffs_vgetf(MountPoint *mp, ino_t ino, int flags, vnode **vpp, int ffs_flags)
 			// previously the source here), returns uint32_t.
 			// lrand() returns int32_t, which isn't ideal but it's
 			// all we have right now
-			ip->i_gen = lrand();
+			// TODO HARVEY We need a u64 random number generator
+			ip->i_gen = -INT32_MIN + rand();
```

Comment:
Talk with Rafa (@fuchicar) for it. I think he included it when made the actual random device removing the old from plan9.